### PR TITLE
API Chat and Marketplace Assistant Improvements

### PR DESCRIPTION
**Pull request by ashera96**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 1]
```diff
@@ -10391,57 +10392,24 @@ public static String getScopesAsString(Set<Scope> scopes) {
...
-        APIManagerConfiguration config = ServiceReferenceHolder.getInstance().
-                getAPIManagerConfigurationService().getAPIManagerConfiguration();
-        String endpoint = config.getFirstProperty(endpointConfigName);
-        String authToken = config.getFirstProperty(authTokenConfigName);
         try {
             HttpPost preparePost = new HttpPost(endpoint + resource);
             preparePost.setHeader(APIConstants.API_KEY_AUTH, authToken);
             preparePost.setHeader(HttpHeaders.CONTENT_TYPE, APIConstants.APPLICATION_JSON_MEDIA_TYPE);
-            preparePost.setHeader("x-request-id", requestId);
+            preparePost.setHeader("apiChatRequestId", requestId);
```
Comment (PasanT9) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:10412]: Let's avoid using magic strings.

Comment (PasanT9) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:10412]: Please check other places as well.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java [hunk 2]
```diff
@@ -3317,20 +3322,32 @@ public String getOpenAPIDefinition(String apiId, String organization) throws API
...
+            JsonNode openAPIDefinitionJsonNode = objectMapper.readTree(getOpenAPIDefinition(apiId, organization));
+            ObjectNode payload = objectMapper.createObjectNode();
+            payload.set("openapi", openAPIDefinitionJsonNode);
+
+            ApiChatConfigurationDTO configDto = ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService()
+                    .getAPIManagerConfiguration().getApiChatConfigurationDto();
+            return APIUtil.invokeAIService(configDto.getEndpoint(), configDto.getAccessToken(),
+                    configDto.getPrepareResource(), payload.toString(), apiChatRequestId);
+        } catch (JsonProcessingException e) {
+            String error = "Error while parsing OpenAPI definition to JSON";
```
Comment (AnuGayan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java:3347]: Let's include some sort of an identifier to identify the API

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/notifier/MarketplaceAssistantApiPublisherNotifier.java [hunk 3]
```diff
@@ -167,11 +164,11 @@ private void deleteRequest(APIEvent apiEvent) throws NotifierException {
 
         try {
             String uuid = apiEvent.getUuid();
-            APIUtil.DeleteApi(marketplaceAssistantConfigurationDto.getEndpoint(),
+            APIUtil.deleteApi(marketplaceAssistantConfigurationDto.getEndpoint(),
                     marketplaceAssistantConfigurationDto.getAccessToken(), marketplaceAssistantConfigurationDto.getApiDeleteResource(), uuid);
         } catch (APIManagementException e) {
-            String errorMessage = "Error encountered while Deleting the API from the vector database";
-            log.error(errorMessage);
+            String errorMessage = "Error encountered while Deleting the API from the vector database" + e.getMessage();
```
Comment (AnuGayan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/notifier/MarketplaceAssistantApiPublisherNotifier.java:170]: Let's add the API name. or some identifier. Please check other places as well

**Review by AnuGayan — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java (hunk 2 above)
Comment (ashera96) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java:3347]: Fixed with https://github.com/wso2/carbon-apimgt/pull/12385/commits/e23b99dea51b29c978fed6b6fefb746969f086e8

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/internal/APIManagerComponent.java [hunk 4]
```diff
@@ -18,6 +18,7 @@
 package org.wso2.carbon.apimgt.impl.internal;
 
 import org.apache.axis2.engine.ListenerManager;
+import org.apache.commons.lang3.StringUtils;
```
[the remark this answers is not in the document]
Comment (ashera96) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/internal/APIManagerComponent.java:21]: Fixed with https://github.com/wso2/carbon-apimgt/pull/12385/commits/e23b99dea51b29c978fed6b6fefb746969f086e8

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/notifier/MarketplaceAssistantApiPublisherNotifier.java (hunk 3 above)
Comment (ashera96) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/notifier/MarketplaceAssistantApiPublisherNotifier.java:170]: Fixed with https://github.com/wso2/carbon-apimgt/pull/12385/commits/e23b99dea51b29c978fed6b6fefb746969f086e8

**Pull request merged**